### PR TITLE
Performance improvements in small data regimes (n < 2000)

### PR DESCRIPTION
**Pull request by jacobrgardner**
In the past few months a lot of our speed optimizations for exact GPs have drifted towards favoring large data regimes (n > 100000). This PR makes two optimizations that achieve around a 2.5x speed up in lower data regimes when using CG for training:
1. Preconditioning is turned off until n > 2000.
1. Use torch.symeig instead of our own batch_symeig now that it supports batching.

This pushes the value of n for which CG overtakes Cholesky in terms of speed back down to around n=1000.

### CODE REVIEW

File: gpytorch/utils/lanczos.py [hunk 1]
```diff
@@ -160,4 +159,9 @@ def lanczos_tridiag_to_diag(t_mat):
 
     TODO: make the eigenvalue computations done in batch mode.
     """
-    return batch_symeig(t_mat)
+    if t_mat.size(-1) < 64:
+        orig_device = t_mat.device
+        retr = torch.symeig(t_mat.cpu(), eigenvectors=True)
+        return (r.to(orig_device) for r in retr)
```
[the remark this answers is not in the document]
Comment (jacobrgardner) [reply gpytorch/utils/lanczos.py]: Done. Had to return the PD masking we did in `eig` anyways.

File: test/examples/test_kissgp_multiplicative_regression.py [hunk 2]
```diff
@@ -83,13 +84,13 @@ def test_kissgp_gp_mean_abs_error(self):
...
-            self.assertTrue(param.grad is not None)
-            self.assertGreater(param.grad.norm().item(), 0)
-        optimizer.step()
+            for param in gp_model.parameters():
+                self.assertTrue(param.grad is not None)
+                self.assertGreater(param.grad.norm().item(), 0)
+            for param in likelihood.parameters():
+                self.assertTrue(param.grad is not None)
+                self.assertGreater(param.grad.norm().item(), 0)
+            optimizer.step()
```
Comment (jacobrgardner) [inline test/examples/test_kissgp_multiplicative_regression.py]: Actually training helps...

Comment (Balandat) [reply test/examples/test_kissgp_multiplicative_regression.py]: consider that...
